src/themes/elife/index.ts

```diff
@@ -1,73 +1,25 @@
-import { first, ready, select, text } from '../../util'
-import * as downloads from './downloads'
-import * as socialSharers from './social-sharers'
-import * as references from './references'
-import DateTimeFormat = Intl.DateTimeFormat
-
-const dateFormatter = new DateTimeFormat('en-US', {
-  month: 'short',
-  day: 'numeric',
-  year: 'numeric',
-})
-
-const formatDate = (dateEl: Element | null): void => {
-  if (dateEl instanceof Element) {
-    const date = new Date(dateEl.innerHTML)
-    dateEl.innerHTML = dateFormatter.format(date)
-  }
-}
-
-const normaliseWhitespace = (txt: string): string => {
-  return txt.replace(/\n/, ' ').replace(/ \s+|\n+/g, ' ')
-}
-
-const getNormalisedTextFromElement = (selector: string): string => {
-  const target = first(selector)
-  if (target !== null) {
-    const sourceText = text(target)
-    if (sourceText !== null) {
-      return normaliseWhitespace(sourceText)
-    }
-  }
-  return ''
-}
-
-const getArticleId = (): string => {
-  return getNormalisedTextFromElement(
-    ':--identifier meta[content="https://registry.identifiers.org/registry/publisher-id"] ~ [itemprop="value"]'
-  )
-}
-
-const getArticleDoi = (): string => {
-  return getNormalisedTextFromElement(
-    ':--identifier meta[content="https://registry.identifiers.org/registry/doi"] ~ [itemprop="value"]'
-  )
-}
-
-const getArticleTitle = (): string => {
-  return getNormalisedTextFromElement(':--title')
-}
+import { first, ready, select } from '../../util'
+import * as dateFormatter from './lib/dateFormatter'
+import * as dataProvider from './lib/dataProvider'
+import * as downloads from './lib/downloads'
+import * as socialSharers from './lib/socialSharers'
+import * as references from './lib/references'
 
 ready((): void => {
-  formatDate(first(':--datePublished'))
+  dateFormatter.format(first(':--datePublished'))
 
   downloads.build(
-    getArticleId(),
+    dataProvider.getArticleId(),
     first(':--title')?.getAttribute('content') ?? ''
```

Comment:
Consolidate data access with https://github.com/stencila/thema/pull/157/files#diff-f28cb2a39f1904fe2a99f06f90e7f9d5R20

Reply:
Addressed in https://github.com/stencila/thema/pull/157/commits/5966ea5d09cd3da3e7ea6b99a00f9ef2d09aa49d